### PR TITLE
Add com.valvesoftware.Steam.Utility.steamtinkerlaunch

### PR DESCRIPTION
**Pull request by HanPrower**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/frostworx/steamtinkerlaunch/issues/27#issuecomment-1125430691
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### As an extra note
Steam Tinker Launcher, when installed normally on a system, can be added to Steam's Compatibility Tools list via a simple command (`steamtinkerlaunch compat add` - documented here: https://github.com/frostworx/steamtinkerlaunch/wiki/Steam-Compatibility-Tool) and whilst the same is possible when using this flatpak the command is "clunky" for an end-user (`flatpak run --command="/app/utils/steamtinkerlaunch/bin/steamtinkerlaunch" --file-forwarding com.valvesoftware.Steam compat add`), as such there is an open issue (https://github.com/flathub/com.valvesoftware.Steam/issues/900) on the Steam flatpak repo about a possible way for a Utility to easily also add a Compatibility Tool.

This shouldn't stop the acceptance of this Utility, it's just something that would be useful to have a discussion on 🙂

### CODE REVIEW

**Comment by TheEvilSkeleton** [comment]
I firmly think that we should really do something about https://github.com/flathub/com.valvesoftware.Steam/issues/900. Sadly no maintainers have responded yet.

**Comment by barthalion** [comment]
Can you poke Steam maintainers on Matrix Flatpak channel?

**Comment by barthalion** [comment]
Either way, I'm fine with merging as it is if you are.

**Comment by barthalion** [comment]
bot, build com.valvesoftware.Steam.Utility.steamtinkerlaunch

**Comment by HanPrower** [comment]
bot, build com.valvesoftware.Steam.Utility.steamtinkerlaunch

**Comment by HanPrower** [comment]
bot, build com.valvesoftware.Steam.Utility.steamtinkerlaunch

**Comment by HanPrower** [comment]
> Either way, I'm fine with merging as it is if you are.

@barthalion It'd be great to get it merged as is for now. It can always be changed or repacked in the future.

**Comment by TheEvilSkeleton** [comment]
@HanPrower I want to chime in for a second. Implementing an extension point takes very little time. I think it's best to wait because otherwise we'll have to deprecate this flatpak, create a new one and submit it.

If you're okay with waiting longer, I can ping the Steam flatpak maintainers and ask them for a new extension point.

If you want to rush, then I won't stop you, but I think it's better for an end user and maintainer perspective to just wait a bit for a better packaging.

**Comment by HanPrower** [comment]
@TheEvilSkeleton Sure, we can wait to see what the devs say (and yes, I am a bit impatient - I hate seeing things just sit ^.^;;).

**Comment by TheEvilSkeleton** [comment]
> Can you poke Steam maintainers on Matrix Flatpak channel?

I did that earlier today, but it doesn't seem like the maintainers have responded. I noticed that they have read messages after mine. I'm not sure if I should ping them again.

**Comment by HanPrower** [comment]
@TheEvilSkeleton We're a week on have you managed to get any news about the issue?

**Comment by TheEvilSkeleton** [comment]
@HanPrower btw, here's the log https://github.com/flathub/com.valvesoftware.Steam/issues/900#issuecomment-1135241741.

**Comment by HanPrower** [comment]
bot, build com.valvesoftware.Steam.Utility.steamtinkerlaunch

**Comment by HanPrower** [comment]
@gasinvein Oh, was waiting on updating this waiting for that PR :sweat_smile: 

That new build installs the `compatibilitytool.vdf` as `SteamTinkerLaunch.vdf` alongside the tool as the Valve specifications for tools state.

**Comment by gasinvein** [comment]
> That new build installs the `compatibilitytool.vdf` as `SteamTinkerLaunch.vdf` alongside the tool as the Valve specifications for tools state.

Right, that's why I didn't notice the manifest; ignore my previous comment.

So, did you test SteamTinkerLaunch with that Steam flatpak PR?
If it woks, maybe it worth upstreaming the `.vdf` files?

**Comment by barthalion** [comment]
So, is it good to merge now?

**Comment by HanPrower** [comment]
@barthalion https://github.com/flathub/com.valvesoftware.Steam/pull/908 is good to merge, yes. It works as intended. Just need someone in charge of the Steam Flatpak to accept it, such as @gasinvein

As soon as that is merged to Steam then this SteamTinkerLaunch PR can finally be accepted.

**Comment by gasinvein** [comment]
> Just need someone in charge of the Steam Flatpak to accept it, such as @gasinvein

I'm not really the one in charge of the Steam flatpak, but anyway, merged it to `beta` branch, should become available in the flathub-beta remote soon.
Also [proposed](https://github.com/flathub/com.valvesoftware.Steam/pull/911) to land everything to `stable`, but since there is a bunch of other changes, it probably needs @nanonyme consent to merge.

File: com.valvesoftware.Steam.Utility.steamtinkerlaunch.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  - name: steamtinkerlaunch
+    build-options:
+      env:
+        PREFIX: /app/utils/steamtinkerlaunch
+    buildsystem: simple
+    build-commands:
+      - make install
+      - install -Dm644 -t ${FLATPAK_DEST}/share/steam/compatibilitytools.d SteamTinkerLaunch.vdf
+      - install -Dm644 -t ${FLATPAK_DEST}/bin toolmanifest.vdf
```
Comment (gasinvein) [inline com.valvesoftware.Steam.Utility.steamtinkerlaunch.yml:43]: Nitpick: installing non-executable files to `$bindir` isn't very nice. Maybe put `toolmanifest.vdf` to extension mount point root (then I guess we'll need to remove `bin/` prefix from `SteamTinkerLaunch.vdf` and add it to `toolmanifest.vdf` instead)?

Comment (HanPrower) [reply com.valvesoftware.Steam.Utility.steamtinkerlaunch.yml:43]: The Valve compat tools documentation state that the `toolmanfiest.vdf` should be next to the tool being lanched. Unless I understood them wrong?

Comment (gasinvein) [reply com.valvesoftware.Steam.Utility.steamtinkerlaunch.yml:43]: If "next to" means "in the same directory", then I wonder if compat tools are supposed to be installed to system `PATH` at all.

**Comment by HanPrower** [comment]
bot, build com.valvesoftware.Steam.Utility.steamtinkerlaunch

**Comment by HanPrower** [comment]
@barthalion This is mergeable now :tada:

**Comment by TheEvilSkeleton** [comment]
@HanPrower did you test it on your end?

**Comment by HanPrower** [comment]
> @HanPrower did you test it on your end?

Yup.

**Comment by TheEvilSkeleton** [comment]
I see. It wasn't working on my end, but I was busy with other stuff so I didn't have the time to provide log. I'll provide log tomorrow if I can.

**Comment by TheEvilSkeleton** [comment]
Okay, I just tested again with a clean Steam flatpak install and it's working. Forget everything I said.

**Pull request closed without merging**